lib/decanter/core.rb

```diff
@@ -67,11 +69,30 @@ def handle_empty_args
       end
 
       def any_inputs_required?
-        handlers.any? do |handler|          
-          handler[1].try(:[], :options).try(:[], :required)
+        handlers.any? do |handler|     
+          return false unless handler.last && handler.last[:options]
+          handler.last[:options][:required]
         end
       end
 
+      def required_inputs
+        handlers.map do |h|
+          options = h.last[:options]
+          return nil unless options && options[:required]
```

Comment:
Is it possible to `return` from a block? I usually see a `LocalJumpError: unexpected return` when I try to do this.